SpatialGDK/Source/SpatialGDK/Private/Tests/WorkingSetSystemTests.cpp

```diff
@@ -0,0 +1,387 @@
+		{
+			Fixture.GetTargetView().AddEntity(MemberEntityId);
+		}
+
+		Fixture.Advance(EmptySet);
+
+		// Expect entities in view not to be set members.
+		for (const Worker_EntityId_Key MarkerEntityId : MarkerEntityIds)
+		{
+			TestFalse(TEXT("Marker entity isn't in view"), System.GetSet(MarkerEntityId).IsSet());
+		}
+	}
+	{
+		// Multiple marker entities appear, all claiming FilledSet entities.
+		for (const Worker_EntityId_Key MarkerEntityId : MarkerEntityIds)
+		{
+			Fixture.GetTargetView().AddEntity(MarkerEntityId);
+
+			Fixture.GetTargetView().AddOrSetComponent(MarkerEntityId, CreateRequest(1, LeaderEntityId, FilledSet).CreateComponentData());
+			Fixture.GetTargetView().AddOrSetComponent(MarkerEntityId, FWorkingSetMarkerResponse().CreateComponentData());
+			Fixture.GetTargetView().AddOrSetComponent(MarkerEntityId, ComponentData(SpatialConstants::WORKING_SET_MARKER_TAG_COMPONENT_ID));
+		}
+
+		// FilledSet entities are being migrated.
+		Fixture.Advance(FilledSet);
+
+		// Expect no working sets to be formed yet.
+		for (const Worker_EntityId_Key MarkerEntityId : MarkerEntityIds)
+		{
+			TestTrue(TEXT("No entities in set yet"), Fixture.IsActualSetEqual(MarkerEntityId, EmptySet));
+		}
+	}
+
+	// Exit early to avoid crashing in the next step as we're accessing sets without checking if they exist.
+	if (HasAnyErrors())
+	{
+		return false;
+	}
+	{
+		// All entities have been migrated.
+		Fixture.Advance(EmptySet);
+
+		// Expect a single working set to be formed as sets were requesting same entities.
+		int32 TotalWorkingSets = 0;
+		for (const Worker_EntityId_Key MarkerEntityId : MarkerEntityIds)
+		{
+			TotalWorkingSets += System.GetSet(MarkerEntityId)->ConfirmedEntities.Num() != 0 ? 1 : 0;
+
+			TestEqual(TEXT("Local view updated with correct epoch"), Fixture.GetWorkingSetResponse(MarkerEntityId).Epoch, 1LL);
+			TestTrue(TEXT("Local view updated with correct epoch"),
+					 AreSetsEqual(Fixture.GetWorkingSetResponse(MarkerEntityId).MemberEntities,
+								  System.GetSet(MarkerEntityId)->ConfirmedEntities));
+		}
+		TestEqual(TEXT("Only one working set approved"), TotalWorkingSets, 1);
+	}
+
+	return true;
+}
+
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FWorkingSetSystemConflictResolution3, "SpatialGDK.WorkingSets.ConflictResolution3",
+								 EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::EngineFilter)
+
+bool FWorkingSetSystemConflictResolution3::RunTest(const FString& Parameters)
+{
+	FWorkingSetSystemTestFixture Fixture;
+	const FWorkingSetSystem& System = Fixture.GetSystem();
+
+	constexpr Worker_EntityId LeaderEntityId = 100;
+	constexpr Worker_EntityId MarkerEntityIds[] = { 1, 2 };
+	const TSet<Worker_EntityId_Key> EmptySet;
+	const TSet<Worker_EntityId_Key> FilledSet{ 3 };
+	const TSet<Worker_EntityId_Key> UpdatedSets[]{ { 3, 4 }, { 4 } };
+
+	const FWorkingSetSystemTestResult::FOutcome Outcome1{ { MarkerEntityIds[0], { UpdatedSets[0], UpdatedSets[0], LeaderEntityId } },
```

Comment:
Should this be UpdatedSets[0] and UpdatedSets[1]?

Reply:
Turns out it actually shouldn't; the FOutcome struct specifies per-set outcomes as "MarkerEntityId, TargetMembers, ConfirmedMembers, LeaderEntityId", so having UpdatedSets[1] two times here makes sense.